source/write-operations/replace-documents.txt

```diff
@@ -0,0 +1,212 @@
+to modify the ``replaceOne()`` method. All methods are optional.
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``bypassDocumentValidation(Boolean bypassDocumentValidation)``
+     - | Specifies whether the replace operation bypasses document validation. This lets you 
+         update documents that don't meet the schema validation requirements, if any 
+         exist. For more information about schema validation, see :manual:`Schema
+         Validation </core/schema-validation/#schema-validation>` in the MongoDB
+         Server manual.
+      
+   * - ``collation(Collation collation)``
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the MongoDB Server manual.  
+
+   * - ``comment(BsonValue comment)``
+     - | Attaches a ``BsonValue`` comment to the operation. For more information, see the :manual:`insert command
+         fields </reference/command/insert/#command-fields>` guide in the
+         MongoDB Server manual.  
+
+   * - ``comment(String comment)``
+     - | Attaches a ``String`` comment to the operation. For more information, see the :manual:`insert command
+         fields </reference/command/insert/#command-fields>` guide in the
+         MongoDB Server manual.  
+
+   * - ``hint(Bson hint)``
+     - | Sets the index for the operation as a ``Bson`` value. 
+         For more information, see the :manual:`hint statement </reference/command/update/#std-label-update-command-hint>`
+         in the MongoDB Server manual.
+
+   * - ``hintString(String hint)``
+     - | Sets the index for the operation as a ``String`` value.
+         For more information, see the :manual:`hint statement </reference/command/update/#std-label-update-command-hint>`
+         in the MongoDB Server manual.
+
+   * - ``let(Bson variables)``
+     - | Specifies a map of parameter names and values. Values must be constant or closed
+         expressions that don't reference document fields. For more information,
+         see the :manual:`let statement
+         </reference/command/update/#std-label-update-let-syntax>` in the
+         MongoDB Server manual.
+
+   * - ``upsert(Boolean upsert)``
+     - | Specifies whether the replace operation performs an upsert operation if no 
+         documents match the query filter. For more information, see the :manual:`upsert
+         statement </reference/command/update/#std-label-update-command-upsert>`
+         in the MongoDB Server manual.
+
+Example
+~~~~~~~
+
+The following code uses the ``replaceOne()`` method to replace a document in the
+``restaurants`` collection. It also sets the ``upsert(true)`` option
+to bypass document-level validation. Because the ``upsert`` option is
+set to ``true``, the driver inserts a new document if the query filter doesn't 
+match any existing documents.
```

Comment:
I: this description is inaccurate
```suggestion
``restaurants`` collection. It also sets the ``upsert(true)`` option
so that the driver inserts a new document if the query filter doesn't 
match any existing documents.
```